### PR TITLE
NIAD-1190: Add support for NHS002 observation range units

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
## Description

Add an optional "units" field to `RangeDetail` for NHS002 messages. The `ObservationMapper` will add it, if given, to each of the low and high range values.

## Jira Ticket

https://gpitbjss.atlassian.net/browse/NIAD-1190

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [x] Acceptance Criteria met
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] Self-reviewed your code
- [x] Code reviewed from two other developers
- [x] If your pull request depends on any other, please link them in the description
- [x] main branch is passing/green on CI
- [x] Add/update any relevant documentation

### CODE REVIEW

**Review by n-r-b — APPROVED** [review]
Looks good 👍

**Review by antonlebedjko — DISMISSED** [review]
LGTM

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/RangeDetailTest.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 class RangeDetailTest {
     @Test
     void testGetKey() {
-        assertThat(new RangeDetail(null, null).getKey()).isEqualTo("RND");
+        assertThat(new RangeDetail(null, null, null).getKey()).isEqualTo("RND");
```
Comment (matthewdoreilly) [inline src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/RangeDetailTest.java:16]: Minor thing, but test data would look a bit better here if we actually gave proper values, like
`new RangeDetail(new BigDecimal("1"), new BigDecimal("10"), "g/dl")`

Comment (andrewwa-kainos) [reply src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/RangeDetailTest.java:16]: For `getKey` the values don't matter - which I'm expressing by using null (and empties and zero etc in other classes). If we put values there may be a false impression that they impact the expectation somehow. What do you think?

**Review by bartek-sarul — APPROVED** [review]

**Review by n-r-b — APPROVED** [review]

**Review by matthewdoreilly — APPROVED** [review]

**Pull request merged**